File: features/authority/authority/Authority.go
package authority

import (
	"github.com/hailongz/golang/db"
)

type Authority struct {
	db.Object
	Uid     int64       `json:"uid" name:"uid" title:"用户ID" index:"ASC"`
	RoleId  int64       `json:"roleId" name:"roleid" title:"角色ID" index:"ASC"`
	ResId   int64       `json:"resId" name:"resid" title:"资源ID" index:"ASC"`
	Options interface{} `json:"options,omitempty" name:"options" title:"其他选项 JSON 叠加" length:"-1"`
}

func (O *Authority) GetName() string {
	return "authority"
}

func (O *Authority) GetTitle() string {
	return "权限"
}

File: features/authority/ts/Authority.ts
import { int64, int32 } from "./lib/less";

/**
 * 权限
 * @type db
 */
export class Authority {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 用户ID
     * @index ASC
     */
    uid: int64 = 0

    /**
     * 角色ID
     * @index ASC
     */
    roleId: int64 = 0

    /**
     * 资源ID
     * @index ASC
     */
    resId: int64 = 0

    /**
     * 其他选项 JSON 叠加
     * @length -1
     */
    options: any

}